coalib/nestedlib/NlInfoExtractor.py

```diff
@@ -0,0 +1,271 @@
+import os
+import logging
+from copy import deepcopy
+from coalib.bears.BEAR_KIND import BEAR_KIND
+from coalib.collecting.Collectors import collect_registered_bears_dirs
+from coalib.parsing.Globbing import glob_escape
+from coalib.collecting.Collectors import (
+    collect_bears, collect_bears_by_aspects)
+
+
+SUPPORTED_LANG_COMB = [{"python", "jinja2"}]
+
+def get_temp_file_lang(nl_info_dict, temp_file):
+    """
+    Return the language of the temp created file.
+    """
+    for file, file_info in nl_info_dict['nl_file_info'].items():
+        for lang, temp_file_name in file_info.items():
+            if temp_file == temp_file_name:
+                return lang
+
+def get_orig_file(nl_info_dict, temp_file):
+    """
+    Return the original file name for the temp_file
+    """
+    for file, file_info in nl_info_dict['nl_file_info'].items():
+        for lang, temp_file_name in file_info.items():
+            if temp_file == temp_file_name:
+                return file
+
+def check_lang_support(lang_list):
+    """
+    Check if the languages supplied by the user is supported by the nested
+    language mode of coala.
+
+    :param lang_list: List containing all the nested languages specified by the
+                      user.
+    """
+    lang_supported = False
+    lang_list = [lang.lower() for lang in lang_list]
+    lang_set = set(lang_list)
+
+    for supported_lang_set in SUPPORTED_LANG_COMB:
+        # The combination of the languages are supported if the difference of
+        # the set is zero.
+        if not supported_lang_set.difference(lang_set):
+            lang_supported = True
+
+    if not lang_supported:
+        logging.error("The language combination are not supported. " +
+                      "Please check if the languages are provided with" +
+                      "the correct names")
+        raise SystemExit(2)
+
+
+def get_bear_dirs(nl_info_dict):
```

Comment:
This looks very similar to Section Class in coalib,
Can you re-use that code than defining your own here?